scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -359,19 +360,23 @@ function Remove-SecurityDaemonResources {
             -ForegroundColor "Red"
         $success = $false
     }
+    else {
+        Write-Verbose "$CmdErr"
+    }
 
     Remove-Item -Recurse $EdgeEventLogMessagesPath -ErrorAction SilentlyContinue -ErrorVariable CmdErr
     if ($?) {
         Write-Verbose "Deleted install directory '$EdgeEventLogMessagesPath'"
     }
     elseif ($CmdErr.FullyQualifiedErrorId -ne "PathNotFound,Microsoft.PowerShell.Commands.RemoveItemCommand") {
+        Write-Verbose "$CmdErr"
         Write-Warning "Could not delete '$EdgeEventLogMessagesPath'."
         Write-Warning "If you're reinstalling or updating IoT Edge, then this is safe to ignore."
         Write-Warning ("Otherwise, please close Event Viewer, or any PowerShell windows where you ran Get-WinEvent, " +
             "then run Uninstall-SecurityDaemon again with '-Force'.")
     }
     else {
-        $success = $false
```

Comment:
This is still needed.

Reply:
With that statement in place, if iotedged is _mostly but not entirely_ uninstalled on a user's system and they call `Uninstall-SecurityDaemon -Force`, then `Remove-SecurityDaemonResources` would return `$false` and the uninstall operation would effectively be considered a failure, even if it managed to completely uninstall the remaining bits of iotedged.

Why is it needed?

Reply:
As discussed offline, yes you're right. I broke this in a135bdfb35fc3a163ebeaf7cb211052fa0410a16; it should not set `$success` to `$false` in that branch.

`Uninstall-SecurityDaemon` doesn't print any error when `Remove-SecurityDaemonResources` returns `$false`; it just doesn't print the success message. That's probably why I didn't notice it when I broke it...